File: 1122-relative-sort-array/1122-relative-sort-array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        vector<int> res;
        map<int,int> z ;
        set <int> x ;
        for(auto &i : arr1) {
            z[i]++;
            x.insert(i);
        }
        vector<int> y ;
        for(auto &i :  arr2){
            if(find(x.begin(),x.end(),i) != x.end()){
                for(int j = 0 ; j <z[i];j++){
                    res.push_back(i);
                }
            }
           
            }
        for(auto &i : x ){
            if(find(res.begin(),res.end(),i) == res.end()){
                for(int j = 0 ; j <z[i];j++){
                    res.push_back(i);
            }
        }
        }
        return res;
    }
};

File: 1122-relative-sort-array/1122-relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        res = {}
        for i in range(len(arr2)):
            res[arr2[i]] = i
        
        helper =  Counter(arr1)
        arr = list(set(arr1))
        ans = []
        rest = []
        for i in arr1:
            if not i in res:
                rest.append(i)
        
        for i in res : 
            ans.append(i)
            for j in range(helper[i]-1):
                ans.append(i)
        rest.sort()
        ans.extend(rest)
        return ans 
            
        
              